# Examining Racial Discrimination in the US Job Market

### Background
Racial discrimination continues to be pervasive in cultures throughout the world. Researchers examined the level of racial discrimination in the United States labor market by randomly assigning identical résumés to black-sounding or white-sounding names and observing the impact on requests for interviews from employers.

### Data
In the dataset provided, each row represents a resume. The 'race' column has two values, 'b' and 'w', indicating black-sounding and white-sounding. The column 'call' has two values, 1 and 0, indicating whether the resume received a call from employers or not.

Note that the 'b' and 'w' values in race are assigned randomly to the resumes when presented to the employer.

### Exercises
You will perform a statistical analysis to establish whether race has a significant impact on the rate of callbacks for resumes.

Answer the following questions **in this notebook below and submit to your Github account**. 

   1. What test is appropriate for this problem? Does CLT apply?
   2. What are the null and alternate hypotheses?
   3. Compute margin of error, confidence interval, and p-value.
   4. Write a story describing the statistical significance in the context or the original problem.
   5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

You can include written notes in notebook cells using Markdown: 
   - In the control panel at the top, choose Cell > Cell Type > Markdown
   - Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet


#### Resources
+ Experiment information and data source: http://www.povertyactionlab.org/evaluation/discrimination-job-market-united-states
+ Scipy statistical methods: http://docs.scipy.org/doc/scipy/reference/stats.html 
+ Markdown syntax: http://nestacms.com/docs/creating-content/markdown-cheat-sheet

****

In [1]:
import pandas as pd
import numpy as np
from scipy import stats

In [3]:
data = pd.io.stata.read_stata('data/us_job_market_discrimination.dta')

In [4]:
pd.set_option('max_columns',None)
data.head()

id ad  education  ofjobs  yearsexp  honors  volunteer  military  empholes  \
0  b  1          4       2         6       0          0         0         1   
1  b  1          3       3         6       0          1         1         0   
2  b  1          4       1         6       0          0         0         0   
3  b  1          3       4         6       0          1         0         1   
4  b  1          3       3        22       0          0         0         0   

   occupspecific  occupbroad  workinschool  email  computerskills  \
0             17           1             0      0               1   
1            316           6             1      1               1   
2             19           1             1      0               1   
3            313           5             0      1               1   
4            313           5             1      1               1   

   specialskills firstname sex race    h    l  call city kind   adid  \
0              0   Allison   f    w  0.0  1.0   0.0    c    a  384.0   
1              0   Kristen   f    w  1.0  0.0   0.0    c    a  384.0   
2              0   Lakisha   f    b  0.0  1.0   0.0    c    a  384.0   
3              1   Latonya   f    b  1.0  0.0   0.0    c    a  384.0   
4              0    Carrie   f    w  1.0  0.0   0.0    c    a  385.0   

   fracblack  fracwhite  lmedhhinc  fracdropout  fraccolp       linc  col  \
0   0.989360   0.005500   9.527484     0.274151  0.037662   8.706325  1.0   
1   0.080736   0.888374  10.408828     0.233687  0.087285   9.532859  0.0   
2   0.104301   0.837370  10.466754     0.101335  0.591695  10.540329  1.0   
3   0.336165   0.637370  10.431908     0.108848  0.406576  10.412141  0.0   
4   0.397595   0.180196   9.876219     0.312873  0.030847   8.728264  0.0   

  expminreq schoolreq  eoe  parent_sales  parent_emp  branch_sales  \
0         5            1.0           NaN         NaN           NaN   
1         5            1.0           NaN         NaN           NaN   
2         5            1.0           NaN         NaN           NaN   
3         5            1.0           NaN         NaN           NaN   
4      some            1.0           9.4       143.0           9.4   

   branch_emp  fed  fracblack_empzip  fracwhite_empzip  lmedhhinc_empzip  \
0         NaN  NaN               NaN               NaN               NaN   
1         NaN  NaN               NaN               NaN               NaN   
2         NaN  NaN               NaN               NaN               NaN   
3         NaN  NaN               NaN               NaN               NaN   
4       143.0  0.0          0.204764          0.727046         10.619399   

   fracdropout_empzip  fraccolp_empzip  linc_empzip  manager  supervisor  \
0                 NaN              NaN          NaN      0.0         1.0   
1                 NaN              NaN          NaN      0.0         1.0   
2                 NaN              NaN          NaN      0.0         1.0   
3                 NaN              NaN          NaN      0.0         1.0   
4            0.070493         0.369903    10.007352      0.0         0.0   

   secretary  offsupport  salesrep  retailsales  req  expreq  comreq  educreq  \
0        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
1        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
2        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
3        0.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   
4        1.0         0.0       0.0          0.0  1.0     1.0     0.0      0.0   

   compreq  orgreq  manuf  transcom  bankreal  trade  busservice  othservice  \
0      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0.0   
1      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0.0   
2      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0.0   
3      1.0     0.0    1.0       0.0       0.0    0.0         0.0         0

In [5]:
# number of callbacks for black-sounding names
sum(data[data.race=='b'].call)

157.0

In [6]:
# number of callbacks for white-sounding names
sum(data[data.race=='w'].call)

235.0

In [7]:
len(data[data.race=='b']), len(data[data.race=='w'])

(2435, 2435)

### 1. What test is appropriate for this problem? Does CLT apply?

The goal of this problem is to understand whether race has a significant impact on the rate of callbacks in US job market or not. For this, we can test the equality of callback rates for both races. Since there are 4870 resumes, the central limit theorem applies and states that the difference between the two proportions is approximately a normal distribution. 

### 2. What are the null and alternate hypotheses?

The null hypothesis is that the callback rate for blacks $p_b$ equals the callback rate for whites $p_w$. The alterative hypothesis is that $p_b$ <> $p_w$.

In [8]:
# Compute the callback rates for both races
n_b = len(data[data.race=='b'])
n_w = len(data[data.race=='w'])
c_b = sum(data[data.race=='b'].call)
c_w = sum(data[data.race=='w'].call)
p_b = c_b / n_b
p_w = c_w / n_w
print('Callback rates for blacks & whites are: {:1.3f}, {:1.3f}'.format(p_b,p_w))

Callback rates for blacks & whites are: 0.064, 0.097


## 3. Compute margin of error, confidence interval, and p-value.

In [9]:
# Compute the standard error for the difference between callback rates
std_err = np.sqrt((p_b*(1-p_b)/n_b)+(p_w*(1-p_w)/n_w))
print('Standard error for the difference between callback rates: {:1.4f}'.format(std_err))

Standard error for the difference between callback rates: 0.0078


In [11]:
# Find 95% and 99% confidence intervals
crit = stats.norm.isf([0.025,0.005])
ci_95 = [p_b-p_w-crit[0]*std_err, p_b-p_w+crit[0]*std_err]
print("95% confidence interval: ({:2.3f}, {:2.3f})".format(ci_95[0],ci_95[1]))

95% confidence interval: (-0.047, -0.017)


In [12]:
ci_99 = [p_b-p_w-crit[1]*std_err, p_b-p_w+crit[1]*std_err]
print("99% confidence interval: ({:2.3f}, {:2.3f})".format(ci_99[0],ci_99[1]))

99% confidence interval: (-0.052, -0.012)


In [13]:
# compute the p-value of one-tail test
z_stat = (p_b - p_w) / std_err
p_value = stats.norm.cdf(z_stat)
print('The p-value for the test of equality of callback rates is: {:1.5f}'.format(p_value))

The p-value for the test of equality of callback rates is: 0.00002


### 4. Write a story describing the statistical significance in the context or the original problem.

Resumes with black-sounding names receive a callback rate of 6.4%, while white names receive a callback rate of 9.7%. This difference of 3.3 percentage points is statistically significant, as the p-value for the test of equality of callback rates is less than 0.001. Moreover, the 99% confience interval suggests that the true callback rate difference could range from 1.2 percent points to 5.2 percent points. Therefore, racial discrmination in the U.S. labor market still appears to be a continued challenge. 

### 5. Does your analysis mean that race/name is the most important factor in callback success? Why or why not? If not, how would you amend your analysis?

The analysis shows that names have a significant effect on callbacks for interviews, but there are also other factors, such as education, years of experience, computer skills and may be even gender that can affect the callback success. A logistic regression is needed to examine which factors are significant predictors and their effect sizes. 
